sycl/source/detail/device_image_impl.hpp

```diff
@@ -220,6 +220,11 @@ class device_image_impl {
       const detail::plugin &Plugin = getSyclObjImpl(MContext)->getPlugin();
       Plugin.call<PiApiKind::piProgramRelease>(MProgram);
     }
+    if (MSpecConstsBuffer) {
+      std::lock_guard<std::mutex> Lock{MSpecConstAccessMtx};
+      const detail::plugin &Plugin = getSyclObjImpl(MContext)->getPlugin();
+      Plugin.call<PiApiKind::piMemRelease>(MSpecConstsBuffer);
```

Comment:
@HabKaffee actually, since https://github.com/intel/llvm/pull/5172 is merged, this should be
```suggestion
      memReleaseHelper(Plugin, MSpecConstsBuffer);
```

Reply:
Got it, I'll merge it and change this line

Reply:
Suggestion applied.